src/mesh/mesh_adaptation.cpp

```diff
@@ -0,0 +1,117 @@
+#include "mesh_adaptation.h"
+
+namespace PHiLiP {
+
+template <int dim, typename real, typename MeshType>
+MeshAdaptation<dim,real,MeshType>::MeshAdaptation(double critical_res_input, int total_ref_cycle, double refine_frac, double coarsen_frac)
+    : critical_residual(critical_res_input)
+    , total_refinement_cycles(total_ref_cycle)
+    , refinement_fraction(refine_frac)
+    , coarsening_fraction(coarsen_frac)
+    , current_refinement_cycle(0)
+    , pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)==0)
+    {}
+
+template <int dim, typename real, typename MeshType>
+void MeshAdaptation<dim,real,MeshType>::adapt_mesh(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
+{
+    if (!refine_mesh)
```

Comment:
In the new commit, adapt_mesh() just adapts the meshes (without any checks) and, since the refine_mesh check is only for the ODE solver (and will not be required outside of that class), it's put back in ODEsolver as mesh_refinement_in_ode_solver.